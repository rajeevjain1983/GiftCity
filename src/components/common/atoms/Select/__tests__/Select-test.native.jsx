// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import SelectBox from '../views/Select.native';
import NativeDropDown from '../../NativeDropDown';

describe('Selectbox component', () => {
  const countryUS = 'United States';
  const displayValue = [
    {
      id: 'US',
      displayName: countryUS,
    },
    { id: 'CA', displayName: 'Canada' },
  ];

  it('renders correctly', () => {
    const props = {
      name: 'test',
      id: 'test',
      heading: 'test',
      input: {
        value: 'test',
        onChange: () => {},
      },
      options: displayValue,
    };
    const component = shallow(<SelectBox {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly for error', () => {
    const props = {
      name: 'test',
      id: 'test',
      heading: 'test',
      input: {
        value: 'test',
        onChange: () => {},
      },
      options: displayValue,
      meta: {
        touched: true,
        error: 'error message',
      },
    };
    const component = shallow(<SelectBox {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('if placeholder is present and input.value is not present then placeholder object is prepended', () => {
    const props = {
      name: 'test',
      id: 'test',
      heading: 'test',
      input: {
        value: '',
        onChange: () => {},
      },
      placeholder: 'placeholder',
      options: displayValue,
      meta: {
        touched: true,
        error: 'error message',
      },
    };
    const component = shallow(<SelectBox {...props} />);
    expect(component.find(NativeDropDown).props().data[0].displayName).toBe(props.placeholder);
  });
});
