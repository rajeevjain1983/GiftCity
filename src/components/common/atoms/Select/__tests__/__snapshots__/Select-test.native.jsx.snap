// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectbox component renders correctly 1`] = `
<Styled(View)
  isAddNewCC={false}
>
  <Styled(Styled(BodyCopy))
    accessibilityText=""
    dataLocator=""
    fontSize="fs10"
    fontWeight="extrabold"
    isFocussed={true}
    margin={null}
    numberOfLines={null}
    text="test"
    textDecoration={null}
  />
  <NativeDropDown
    androidPickerStyle={null}
    bottomBorderOnly={true}
    buttonVariation="mobileApp-select"
    data={
      Array [
        Object {
          "displayName": "United States",
          "id": "US",
        },
        Object {
          "displayName": "Canada",
          "id": "CA",
        },
      ]
    }
    disabled={false}
    iconRight={false}
    id="test"
    lightGrayColor={false}
    name="test"
    onValueChange={[Function]}
    pickerStyleObj={Object {}}
    selectedValue="test"
    textAlignLeft={false}
    width="100%"
  />
</Styled(View)>
`;

exports[`Selectbox component renders correctly for error 1`] = `
<Styled(View)
  isAddNewCC={false}
>
  <Styled(Styled(BodyCopy))
    accessibilityText=""
    dataLocator=""
    fontSize="fs10"
    fontWeight="extrabold"
    isFocussed={true}
    margin={null}
    numberOfLines={null}
    text="test"
    textDecoration={null}
  />
  <NativeDropDown
    androidPickerStyle={null}
    bottomBorderOnly={true}
    buttonVariation="mobileApp-select"
    data={
      Array [
        Object {
          "displayName": "United States",
          "id": "US",
        },
        Object {
          "displayName": "Canada",
          "id": "CA",
        },
      ]
    }
    disabled={false}
    iconRight={false}
    id="test"
    lightGrayColor={false}
    name="test"
    onValueChange={[Function]}
    pickerStyleObj={Object {}}
    selectedValue="test"
    textAlignLeft={false}
    width="100%"
  />
  <Styled(View)>
    <Styled(View)
      spacingStyles="margin-right-XS"
    >
      <Styled(ImageComp)
        alt=""
        cacheType="immutable"
        height="14px"
        host=""
        isFastImage={false}
        onError={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        priority="normal"
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/images/alert-triangle.png",
          }
        }
        url=""
        width="16px"
      />
    </Styled(View)>
    <Styled(BodyCopy)
      accessibilityText=""
      color="error"
      dataLocator=""
      fontFamily="secondary"
      fontSize="fs12"
      fontWeight="extrabold"
      margin={null}
      numberOfLines={null}
      text="error message"
      textDecoration={null}
    />
  </Styled(View)>
</Styled(View)>
`;
